### DISCUSSION TITLE
Why is Maximize/Unmaximize/Minimize Behavior Different when Window Decorations are Off?

### DISCUSSION THREAD

**Discussion by e-l-i-j-a-h** [tauri-apps/tauri#2064] [Q&A] [answered]
I wanted a custom title bar for my app, so I started by setting the main window's decorations to `false` in `tauri.conf.json` to hide the default bar and borders. After I built my own title bar, I noticed that clicking my "maximize" button would make the window cover the taskbar. I also noticed that there was no animation when maximizing, unmaximizing, or minimizing the window. I assumed that firing max/unmax/min using Javascript just didn't do the animation. However, when I tried my buttons again with decorations enabled, the animations returned and maximizing did not cover the taskbar.

Is there a way to retain the window animations and correct maximize behavior while leaving the default title bar hidden?

Another feature I just remembered was missing was the ability to resize my window by dragging the mouse on the window border, even though I have the `resizable` attribute set to `true`. I guess the window border is what allows the window to be resized, which is absent when decorations are disabled.

I am using Windows 10 if this helps.

**Update on Ubuntu 20.04:** Maximizing does not cover the taskbar, but my max/unmax button does not do anything until I move my mouse, click, or some other mouse input is received. It seems other promise-based functions such as `setIcon` don't resolve until I move my mouse either, even when invoked using the console. Strange.

**Comment by amrbashir** [comment, marked as answer]
For the windows part we are aware of this issue and hopefully I can fix it some time this week, you can subscribe to these issues to get notified when they are fixed:
https://github.com/tauri-apps/tao/issues/104
https://github.com/tauri-apps/tao/issues/103

As for the linux part, might be worth to open an issue.

**Reply by e-l-i-j-a-h** [reply]
Thanks, I will check those out.

**Reply by NitinRamnani** [reply]
@amrbashir Issue mentioned by @literal-line regarding maximize behavior occurs on Mac also.

**Reply by amrbashir** [reply]
I think that's the correct behaviour on macOS, join our discord and our macOS experts will answer.

**Comment by FabianLars** [comment]
As for the linux issue. Are you using the new beta versions which got released a few hours ago? This behaviour should be fixed afaik.

**Reply by e-l-i-j-a-h** [reply]
Thank you for telling me this! The new update fixed the Linux issues, including another bug I noticed where the the icon didn't show up in the taskbar. Updating Tauri didn't even cross my mind since I only installed it on this old Linux machine one day ago.

This is what I like about Tauri's development. Whenever I find a new bug, I know within a few days it will probably be fixed.
